File: solutions/338-counting-bits/counting-bits.php
class Solution {

    /**
     * @param Integer $n
     * @return Integer[]
     */
    function countBits($n) {
    $ans = array_fill(0, $n + 1, 0);
    for ($i = 1; $i <= $n; $i++) {
        $ans[$i] = $ans[$i >> 1] + ($i & 1);
    }
    return $ans;
    }
}

File: solutions/338-counting-bits/counting-bits.rs
impl Solution {
    fn count_bits(n: i32) -> Vec<i32> {
        let mut ans = vec![0; (n + 1) as usize];
        for i in 1..=n {
            ans[i as usize] = ans[(i >> 1) as usize] + (i & 1);
        }
        ans
    }
}